File: common/src/main/java/com/ssblur/scriptor/helpers/targetable/ItemTargetable.java
package com.ssblur.scriptor.helpers.targetable;

import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

public class ItemTargetable extends EntityTargetable {
  ItemStack itemStack;
  boolean targetItemByDefault;

  public ItemTargetable(ItemStack itemStack, Player entity) {
    super(entity);
    this.itemStack = itemStack;
    this.targetItemByDefault = true;
  }

  public ItemStack getTargetItem() {
    return itemStack;
  }

  public boolean shouldTargetItemByDefault() { return targetItemByDefault; }

  public ItemTargetable withTargetItemByDefault(boolean targetItemByDefault) {
    this.targetItemByDefault = targetItemByDefault;
    return this;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/helpers/targetable/ItemTargetable.kt
package com.ssblur.scriptor.helpers.targetable

import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack

open class ItemTargetable(var targetItem: ItemStack, entity: Player?): EntityTargetable(
  entity!!
) {
  var targetItemByDefault: Boolean = true

  fun shouldTargetItem(): Boolean {
    return targetItemByDefault
  }

  fun withTargetItem(targetItemByDefault: Boolean): ItemTargetable {
    this.targetItemByDefault = targetItemByDefault
    return this
  }
}
